src/evict/evict_internal.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+#include "wt_prelude.h"
+extern void __wti_evict_list_clear_page(WT_SESSION_IMPL *session, WT_REF *ref);
```

Comment:
Breaking `extern.h` into per-component headers removes 2056 lines (80%) from extern.h. Note this PoC is only a partial change. Some files like src/checksum and the `_inline.h headers haven't been touched.